File: src/app/Class/BandConcert.ts
import { Band } from "./Band";
//import { Concert } from './Concert';

export class BandConcert {
  id: number;
  private band: Band;
  // private concert: Concert;

  constructor(obj?: any) {
    this.id = (obj && obj.id) || 0;
    this.band = (obj && obj.band) || null;
    //this.concert = (obj && obj.concert) || null;
  }
  /*
    // Getters

    getId(): number {
        return this.id;
    }

    getBand(): Band {
        return this.band;
    }
    
    getConcert(): Concert {
        return this.concert;
    }

    // Setter

    setBand(band: Band) {
        this.band = band;
    }

    setConcert(concert: Concert) {
        this.concert = concert;
    }
*/
}

File: src/main/java/musiccube/entities/BandConcert.java
package musiccube.entities;

import javax.persistence.*;

@Entity
@Table(name = "band_concert", schema = "music_cube")
public class BandConcert {

    @Id
    @GeneratedValue
    @Column(name = "band_concert_id")
    private int id;

    @ManyToOne
    @JoinColumn(name = "band_id", referencedColumnName = "band_id")
    private Band band;

    @ManyToOne
    @JoinColumn(name = "concert_id", referencedColumnName = "concert_id")
    private Concert concert;

    public BandConcert() {}

    public BandConcert(Band band, Concert concert) {
        this.band = band;
        this.concert = concert;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Band getBand() {
        return band;
    }

    public void setBand(Band band) {
        this.band = band;
    }

    public Concert getConcert() {
        return concert;
    }

    public void setConcert(Concert concert) {
        this.concert = concert;
    }
}
